src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/ComplexFlowValidationResponse.groovy

```diff
@@ -0,0 +1,50 @@
+package org.openkilda.functionaltests.helpers.model
+
+import org.openkilda.northbound.dto.v1.flows.FlowValidationDto
+import org.openkilda.northbound.dto.v1.flows.PathDiscrepancyDto
+import org.openkilda.northbound.dto.v2.haflows.HaFlowValidationResult
+import org.openkilda.northbound.dto.v2.yflows.YFlowValidationResult
+
+import groovy.transform.ToString
+
+@ToString
+class ComplexFlowValidationResponse {
+    private static final String COOKIE_ID_IN_RULE_DISCREPANCY_STRING_REGEX =  /-?\d{19}/
+
+    boolean asExpected
+    List<SubFlowValidationDiscrepancies> subFlowsDiscrepancies
+
+    ComplexFlowValidationResponse(YFlowValidationResult yFlowValidation) {
+        this.asExpected = yFlowValidation.asExpected
+        this.subFlowsDiscrepancies = collectDiscrepanciesPerSubFlow(yFlowValidation.subFlowValidationResults)
+    }
+
+    ComplexFlowValidationResponse(HaFlowValidationResult haFlowValidation) {
+        this.asExpected = haFlowValidation.asExpected
+        this.subFlowsDiscrepancies = collectDiscrepanciesPerSubFlow(haFlowValidation.subFlowValidationResults)
+
+    }
+
+    static private List<SubFlowValidationDiscrepancies> collectDiscrepanciesPerSubFlow(List<FlowValidationDto> subFlowValidationDetails) {
+        List<SubFlowValidationDiscrepancies> discrepanciesPerSubFlow = subFlowValidationDetails.groupBy { it.flowId }.collect{
+            HashMap<FlowDirection, List<PathDiscrepancyDto>> discrepancy = [:]
+            it.value.each {
+                verifyValidateLogic(it)
+                it.asExpected ?: discrepancy.put(FlowDirection.getByDirection(it.direction.toLowerCase()),  it.discrepancies)
```

Comment:
It might be not the case,
 
but is it possible that there are at least two discrepancies  for the same flowid and the same direction?
what if in the following loop :
```
it.value.each {
                verifyValidateLogic(it)
                it.asExpected ?: discrepancy.put(FlowDirection.getByDirection(it.direction.toLowerCase()),  it.discrepancies)
}
```
there will be at least two iteration with the same flowId and the same direction?

Reply:
We interact with List<FlowValidationDto>  response and we use List<PathDiscrepancyDto> discrepancies that should can contain more than one discrepancy per direction.

Reply:
added check for direction per sub-flow (to clarify that we don't have 2 FORWARD validation responses with one discrepancy instead of one FORWARD and two discrepancies) + if we have some invalid direction, it fails during getting FlowDirection